x-pack/platform/plugins/shared/fleet/server/services/agent_policy.ts

```diff
@@ -219,50 +232,95 @@ class AgentPolicyService {
     }
 
     if (!options.skipValidation) {
+      logger.debug(`Validating agent policy [${id}] before update`);
+
       await validateOutputForPolicy(
         soClient,
         agentPolicy,
         existingAgentPolicy,
         getAllowedOutputTypesForAgentPolicy({ ...existingAgentPolicy, ...agentPolicy })
       );
     }
-    await soClient.update<AgentPolicySOAttributes>(savedObjectType, id, {
-      ...agentPolicy,
-      ...(options.bumpRevision ? { revision: existingAgentPolicy.revision + 1 } : {}),
-      ...(options.removeProtection
-        ? { is_protected: false }
-        : { is_protected: agentPolicy.is_protected }),
-      updated_at: new Date().toISOString(),
-      updated_by: user ? user.username : 'system',
-    });
+
+    // If space awareness is enabled and the soClient that was passed on input is not
+    // scoped to the space that the agnet policy is associated with, then initialize
+    // a new client for the update. This should be safe, since the `soClient` provided
+    // on input was used to retrieve the agent policy, thus we know it has access to it.
+    let soClientForUpdate = soClient;
+
+    if (isSpacesEnabled) {
+      const soClientSpaceId = soClient.getCurrentNamespace();
+
+      if (soClientSpaceId && existingAgentPolicy?.space_ids?.includes(soClientSpaceId)) {
+        logger.debug(`Using soClient provided on input for update to agent policy [${id}]`);
+        soClientForUpdate = soClient;
+      } else {
+        const spaceIdForUpdate = existingAgentPolicy?.space_ids?.at(0) ?? DEFAULT_SPACE_ID;
+
+        logger.debug(
+          `Initiating internal soClient scoped to [${spaceIdForUpdate}] for making the update to agent policy [${id}]`
+        );
+        soClientForUpdate = appContextService.getInternalUserSOClientForSpaceId(spaceIdForUpdate);
+      }
+    }
+
```

Comment:
I think this may be an issue, as we could escalate some privileges to user here by mistake. That service probably should rely on the user providing a correctly scoped client. 

In what calls this is needed?

Reply:
I have made so many updates just trying to get things to work that I don't remember now which one required this change. I think the issue was with `bulkUpdate()`... But it might have been a change prior to me making the change in `bulkUpdate` to chunk up the updates by space.

Looks like I'm very close to getting all tests to pass... I'm going to come back here and try to remove this code to see what falls out once I get test to all pass.